---
title: "Plotting Functions"
author: "Chiara Folland"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/00104561/Library/CloudStorage/OneDrive-UWA/Research/Projects/AIM3_TRANSCRIPTOMICS/app/rnawebapp")
library(dplyr)
library(tidyr)
library(ggplot2)
library(OUTRIDER)
library(FRASER)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
library(SGSeq)
```

```{r bamplot, message=FALSE}

```


```{r volcplot, message=FALSE}
OUTRIDER::plotVolcano(ods_hgnc, sampleID = "HP24-0007", xaxis = "zscore", label = aberrant)
FRASER::plotVolcano(fds, sampleID = "TAM-M1475", type = "theta")
plotCountCorHeatmap(ods_hgnc, normalized=TRUE, nRowCluster=4)
plotAberrantPerSample(ods_hgnc, padjCutoff=0.05)
plotExpressionRank(ods_hgnc, "ABCD3", basePlot=TRUE)

```

```{r cortar, message=FALSE}



```

```{r sashimi}
library(dasper)
# Extract exons by transcript
ref <- exonsBy(txdb, by = "tx", use.names = TRUE)

# use GenomicState to load txdb (GENCODE v31)
ref <- GenomicState::GenomicStateHub(version = "31", genome = "hg38", filetype = "TxDb")[[1]]
junctions_processed <- junction_process(junctions_example, ref, types = c("ambig_gene", "unannotated"))

plot_sashimi(junctions = junction_filter(junctions_processed), ref = ref, gene_tx_id = "ENSG00000142156.14", gene_tx_col = "gene_id", sum_func = NULL)

library(SummarizedExperiment)
library(dasper)

junctions <- rowRanges(fds, type = "psi5")
seqlevels(junctions) <- sub("^chr", "", seqlevels(junctions))
seqnames(junctions) <- sub("^chr", "", seqnames(junctions))

# Add metadata
junctions$gene_id <- junctions$hgncSymbol  # or your gene column
junctions$case_id <- "HP23-014"            # sample you're plotting

bam_paths <- colData(fds)$RNA_BAM_FILE
names(bam_paths) <- colData(fds)$sampleID

coverage_paths_case <- bam_paths["HP23-014"]
coverage_paths_control <- bam_paths[c("IVCT-61Y-F")]

txdb <- makeTxDbFromGFF("~/Library/CloudStorage/OneDrive-UWA/Research/Projects/AIM3_TRANSCRIPTOMICS/app/rnawebapp/data/Homo_sapiens.GRCh38.113.gtf", format = "gtf") 

plot_sashimi(
  junctions = junctions,
  ref = txdb,
  gene_tx_id = entrez_id,
  gene_tx_col = "gene_id",
  case_id = "HP23-014",
  coverage_paths_case = coverage_paths_case,
  coverage_paths_control = coverage_paths_control,
  count_label = TRUE
)

```

```{r gene plot, message=FALSE}
#' Generate a faceted plot of gene expression for multiple genes
#'
#' @param ods An OUTRIDER dataset 
#' @param genes A character vector of gene symbols or IDs
#' @param sampleID A sample ID to highlight in red
#' 
#' @return A ggplot object with facets for each gene, highlighting the sample
#' @export
#'
#' @importFrom dplyr filter
#' @importFrom tidyr gather
#' @importFrom ggplot2 ggplot geom_violin geom_boxplot geom_jitter geom_point facet_wrap theme element_blank labs aes

plot_gene_expression_multi <- function(ods, genes, sampleID) {
  # Extract and reshape normalized counts
  counts_norm <- as.data.frame(counts(ods, normalized = TRUE))
  counts_norm$geneID <- rownames(counts_norm)
  
  long_counts <- counts_norm %>%
    tidyr::gather("sample_ID", "counts", -geneID) %>%
    dplyr::filter(geneID %in% genes)
  
  # Highlighted sample
  highlight <- dplyr::filter(long_counts, sample_ID == sampleID)
  other <- dplyr::filter(long_counts, sample_ID != sampleID)
  
  # Plot
  p <- ggplot(other, aes(x = factor(0), y = counts)) +
    facet_wrap(~geneID, scales = "free_y") +
    geom_violin(fill = "white") +
    geom_boxplot(width = 0.3, outlier.shape = NA) +
    #geom_jitter(color = "darkgrey", size = 1, alpha = 0.8) +
    geom_point(data = highlight, aes(x = factor(0), y = counts), color = "red", size = 2) +
    theme(
      axis.text.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.title.x = element_blank()
    ) +
    labs(
      x = NULL,
      y = "Expression (normalized counts)",
      title = paste("Expression for", paste(genes, collapse = ", "))
    )
  
  return(p)
}
genes <- c("ACTA1", "ACTA2", "ABCD3")
plot_gene_expression_multi(ods_hgnc, genes, "OPMD1-13")
```
