# Preparing the Input Data

## Sample Sheet (CSV/TSV)

A samplesheet is required to describe the input samples, their phenotype, the location of BAM files and outputs from the fusion calling.

| RNA_ID | RNA_BAM_FILE | DROP_GROUP | PHENOTYPE | FUSIONS              |
|--------|--------------|------------|-----------|----------------------|
| S01    | /path/to/bam | MUSCLE     | Myopathy  | /path/to/fusions.tsv |
| S02    | /path/to/bam | MUSCLE     | Dystrophy | /path/to/fusions.tsv |

An [example samplesheet]() can be found in the GitHub repository.

## OUTRIDER Dataset

The OUTRIDER Dataset (`ods.Rds`) is an R object that stores the entire OUTRIDER dataset and results.
If you are running DROP, it can be found by navigating to the processed results directory for the `AberrantExpression` module:
For example, `./processed_results/aberrant_expression/v113/outrider/MUSCLE/ods.Rds`

## FRASER Dataset

The FRASER Dataset (`fds.Rds`) is an R object that stores the entire FRASER dataset and results.
If you are running DROP, it can be found by navigating to the processed results directory for the `AberrantSplicing` module:
For example, `./processed_results/aberrant_splicing/datasets/savedObjects/MUSCLE--v113/fds-object.RDS`

The `fds-object.Rds` relies on `.h5` data objects generated by FRASER, which are located in the processed data directory for the `AberrantSplicing` module:
For example, `./processed_data/aberrant_splicing/datasets/savedObjects/MUSCLE`

> ⚠ Use caution when transferring these objects from a remote server to a local server to use as input into the RNAwebapp and use the following rules:

- If using outputs from DROP, place the `.h5` objects in a sub-directory of the data directory called `savedObjects/[DROUP_GROUP]--[GENE_ANNOTATION]`
## BAM Directory

Enter the absolute path to where the BAM files are stored. 
For example: `~/RNAseq/data/bams`

## RNA Variants

## Important Considerations

> ⚠️ Sample IDs must match across all files.

It is important that the sample IDs in the sample annotation match those in the OUTRIDER and FRASER datasets and the RNA VCFs.
This can be confirmed by inspecting the column names of the ods.Rds and fds.Rds files:
```
colnames(ods.Rds)
colnames(fds.Rds)
```
It is okay to have samples present in one dataset but not the other. 

> ⚠ BAM files must be locally accessible.

For the analysis to run properly, the RNA BAM files must be stored in the same directory, located locally.

---
